### PR TITLE
Load example solutions for cached levels async

### PR DESCRIPTION
**Pull request by maureensturgeon**
This is a portion of the full set of changes [here](https://github.com/code-dot-org/code-dot-org/pull/43126). I've split this out so that it's easier to review.

Changes here include:
- Updating load app to use await syntax instead of .then
- Load example solutions async in loadApp to ensure they're set to pageConstants redux for cached levels
- Teacher panel gets example solutions from redux instead of teacher_panel.html.haml

## Testing story
I tested that the example_solutions endpoint works and that example solutions are set to pageConstants.exampleSolutions for cached levels. Also added unit tests.

## Follow-up work
Apply the rest of the changes in https://github.com/code-dot-org/code-dot-org/pull/43126

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/code-studio/components/progress/teacherPanel/TeacherPanel.jsx [hunk 1]
```diff
@@ -299,7 +301,8 @@ export default connect(
       students: state.teacherSections.selectedStudents,
       levelsWithProgress: state.teacherPanel.levelsWithProgress,
       isLoadingLevelsWithProgress:
-        state.teacherPanel.isLoadingLevelsWithProgress
+        state.teacherPanel.isLoadingLevelsWithProgress,
+      exampleSolutions: state.pageConstants?.exampleSolutions
```
Comment (jamescodeorg) [inline apps/src/code-studio/components/progress/teacherPanel/TeacherPanel.jsx:305]: For my education, when is pageConstants null?

File: dashboard/db/schema.rb [hunk 2]
```diff
@@ -515,7 +515,7 @@
     t.index ["library_name", "library_version", "question_name"], name: "index_foorm_library_questions_on_multiple_fields", unique: true
   end
 
-  create_table "foorm_simple_survey_forms", options: "ENGINE=InnoDB DEFAULT CHARSET=utf8 COLLATE=utf8_unicode_ci", force: :cascade do |t|
+  create_table "foorm_simple_survey_forms", options: "ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_unicode_ci", force: :cascade do |t|
```
Comment (jamescodeorg) [inline dashboard/db/schema.rb]: Oops, I think this file got included by accident?

**Review by jamescodeorg — APPROVED** [review]
Looks great, just revert schema.rb.

File: dashboard/db/schema.rb (hunk 2 above)
Comment (maureensturgeon) [reply dashboard/db/schema.rb]: Oops! Yes it did

File: apps/src/code-studio/components/progress/teacherPanel/TeacherPanel.jsx (hunk 1 above)
Comment (maureensturgeon) [reply apps/src/code-studio/components/progress/teacherPanel/TeacherPanel.jsx:305]: `pageConstants` are only set for some pages (I think level pages), so `state.pageConstants` is undefined for a unit overview page like `/s/csd3-2020`. The teacher panel is displayed on this page, but having example solutions doesn't make sense in a unit overview context, which is why this works. 

Maybe displayLevelExamples should also include a check for being on a level page? That may make the logic more explicit.

Comment (jamescodeorg) [reply apps/src/code-studio/components/progress/teacherPanel/TeacherPanel.jsx:305]: That would be good if it's easy to determine that we're on a level page.  If not, maybe a comment would be helpful?

Comment (maureensturgeon) [reply apps/src/code-studio/components/progress/teacherPanel/TeacherPanel.jsx:305]: I'll get this in on one of the following PRs

**Pull request merged**